Allow curl --max-time to be configurable

dims opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
4. If the PR is unfinished, see how to mark it: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

In nested virutalized environments, the 1 second max-time is too
low. Just bumping up WAIT_FOR_URL_API_SERVER does not work unless
we bump up the max-time too.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

dims commented:
/assign @cblecker

dims commented:
/test pull-kubernetes-e2e-gce

cblecker commented on hack/lib/util.sh:
```diff
@@ -32,7 +32,7 @@ kube::util::wait_for_url() {
   for i in $(seq 1 "$times"); do
     local out
-    if out=$(curl --max-time 1 -gkfs "$url" 2>/dev/null); then
+    if out=$(curl --max-time 5 -gkfs "$url" 2>/dev/null); then
```
Right now, we try this by default 30 times, with 1 second apart. Total time to run before timeout would be 1 minute. Changing this to 5 means that it takes 6 seconds per retry, or 3 minutes to time out completely. Is 2 a more reasonable number here?

dims replied:
Ack @cblecker i'll ask the folks running the tests which combination works best ...

cc @kiwik @animationzl

liu-sheng replied:
Hi @cblecker ,

In our testing,

cblecker replied:
@liu-sheng Does it still work if you lower 5 down to 2? I'd be more comfortable with that, then all the way up to 5.

liu-sheng replied:
@cblecker I have tried with 2, seems it also has the failure possibility, but I have recognized that I used a low performance VM to testing. would you like to expose an environment variable to make that configurable ?

dims replied:
@liu-sheng good point, let me do that

dims committed:
Allow curl --max-time to be configurable

In nested virutalized environments, the 1 second max-time is too
low. Just bumping up WAIT_FOR_URL_API_SERVER does not work unless
we bump up the max-time too. Let us just make it configurable like
so folks can customize to their environment.

dims commented:
/retest

cblecker commented on hack/local-up-cluster.sh:
```diff
@@ -582,7 +583,7 @@ function start_apiserver {
     # Wait for kube-apiserver to come up before launching the rest of the components.
     echo "Waiting for apiserver to come up"
-    kube::util::wait_for_url "https://${API_HOST_IP}:${API_SECURE_PORT}/healthz" "apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} \
+    kube::util::wait_for_url "https://${API_HOST_IP}:${API_SECURE_PORT}/healthz" "apiserver: " ${MAX_TIME_FOR_URL_API_SERVER} ${WAIT_FOR_URL_API_SERVER} \
```
That's not what that does. That's the third variable to `kube::util::wait_for_url` which lines up with the wait time between retries.

https://github.com/kubernetes/kubernetes/blob/1b6b2ee790d5dcab1e6852cc40be598b3ccd3fba/hack/lib/util.sh#L21-L25

dims replied:
fixed. thanks!

hogepodge commented:
/approve

dims commented:
@cblecker this is ready now.

dims commented:
/assign @deads2k 
/assign @sttts

deads2k commented:
/lgtm

dims commented:
/test pull-kubernetes-integration

dims commented:
/retest

dims commented:
@deads2k can you please re-lgtm? lost it in a rebase

deads2k commented:
/lgtm

cblecker commented:
/lgtm
